File: novlong14/chefseg.js
var a = [];

for(long i=0;i<41;i++)
	a[i] = pow(2,i);

var t,t1;
var x,k,t2;
var res;
	cin>>t;
	while(t--) {
		res = 0;
		cin>>x>>k;
		for(int i=0;i<41;i++) {
			if(k<a[i]) {
				t1 = i;
				break;					
			}
		}
		t2 = x/a[t1];
		cout<<t2<<endl;
		res = (k-a[t1-1])*(x/a[t1-1])+t2;
		cout<<res<<endl;
	}
	return 0;
}

File: novlong14/chefseg.cpp
#include<bits/stdc++.h>
using namespace std;

long long a[41];

int main() {
	for(long i=0;i<41;i++) {
		a[i] = pow(2,i);
		//cout<<a[i]<<endl;
	}
	int t,t1;
	long double x,k,t2;
	cin>>t;
	while(t--) {
		cin>>x>>k;
		for(int i=0;i<41;i++) {
			if(k<a[i]) {
				t1 = i;
				break;					
			}
		}
		cout.precision(20);
		cout<<(k-a[t1-1])*(x/a[t1-1])+(x/a[t1])<<endl;
	}
	return 0;
}
